exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java

```diff
@@ -0,0 +1,152 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.exporters.otlp;
+
+import io.grpc.ManagedChannel;
+import io.grpc.StatusRuntimeException;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import io.opentelemetry.proto.collector.trace.v1.TraceServiceGrpc;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.ThreadSafe;
+
+/** Exports spans using OTLP via gRPC, using OpenTelemetry's protobuf model. */
+@ThreadSafe
+public final class OtlpGrpcSpanExporter implements SpanExporter {
+  private static final Logger logger = Logger.getLogger(OtlpGrpcSpanExporter.class.getName());
+
+  private final TraceServiceGrpc.TraceServiceBlockingStub blockingStub;
+  private final ManagedChannel managedChannel;
+  private final long deadline;
+
+  /**
+   * Creates a new Jaeger gRPC Span Reporter with the given name, using the given channel.
+   *
+   * @param channel the channel to use when communicating with the Jaeger Collector.
+   * @param deadline max waiting time for the collector to process each span batch. When set to 0 or
+   *     to a negative value, the exporter will wait indefinitely.
+   */
+  private OtlpGrpcSpanExporter(ManagedChannel channel, long deadline) {
+    this.managedChannel = channel;
+    this.blockingStub = TraceServiceGrpc.newBlockingStub(channel);
+    this.deadline = deadline;
+  }
+
+  /**
+   * Submits all the given spans in a single batch to the Jaeger collector.
+   *
+   * @param spans the list of sampled Spans to be exported.
+   * @return the result of the operation
+   */
+  @Override
+  public ResultCode export(List<SpanData> spans) {
+    ExportTraceServiceRequest exportTraceServiceRequest =
+        ExportTraceServiceRequest.newBuilder()
+            .addAllResourceSpans(SpanAdapter.toProtoResourceSpans(spans))
+            .build();
+
+    try {
+      TraceServiceGrpc.TraceServiceBlockingStub stub = this.blockingStub;
+      if (deadline > 0) {
+        stub = stub.withDeadlineAfter(deadline, TimeUnit.MILLISECONDS);
+      }
+
+      // for now, there's nothing to check in the response object
+      // noinspection ResultOfMethodCallIgnored
+      stub.export(exportTraceServiceRequest);
+      return ResultCode.SUCCESS;
+    } catch (StatusRuntimeException e) {
+      // Retryable codes from https://github.com/open-telemetry/oteps/pull/65
+      switch (e.getStatus().getCode()) {
+        case CANCELLED:
+        case DEADLINE_EXCEEDED:
+        case RESOURCE_EXHAUSTED:
+        case OUT_OF_RANGE:
+        case UNAVAILABLE:
+        case DATA_LOSS:
+          return ResultCode.FAILED_RETRYABLE;
```

Comment:
I thought that we had decided that retrying would be the responsibility of the exporter, and that we wouldn't use this code any more. I guess that never got changed, but I still think the exporter should handle its own retries. (also, I don't think any of the span processors currently do any retry logic).

Reply:
Looks like this is still in the spec. I'll open up a spec issue to discuss.

Reply:
https://github.com/open-telemetry/opentelemetry-specification/issues/509

Reply:
As explained in that issue, this is an exporter so then I need to implement retry logic and I will use the same logic to determine when to retry.